lighthouse-core/audits/dobetterweb/doctype.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('../audit');
+
+class Doctype extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      name: 'doctype',
+      description: 'Page has the HTML doctype',
+      failureDescription: 'Page is missing the HTML doctype',
+      helpText: 'Specifying a doctype prevents the browser from switching to quirks-mode.' +
+                'Read more on the ' +
+                '[MDN Web Docs page](https://developer.mozilla.org/en-US/docs/Glossary/Doctype)',
+      requiredArtifacts: ['Doctype'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {LH.Audit.Product}
+   */
+  static audit(artifacts) {
+    if (artifacts.Doctype === null) {
```

Comment:
Just a nitpick here, you can do `if(!artifacts.Doctype)`, and maybe the naming should be lower case?

Reply:
> and maybe the naming should be lower case?

It's uppercase for historical reasons (it automatically uses the the gatherer class name), and we have wanted to change that at some point, but not starting with this PR :)

Reply:
@ev1stensberg I totally could but, I tend to want to be explicit. And as the compiler/minifier will optimise the code in the end, it seems ok. Thoughts?

Reply:
I decided to add your suggestion in the end ;)